### ISSUE TITLE
Inconsistency between handling of block image and inline image

### ISSUE THREAD

**Issue by ysb33r** [asciidoctor/asciidoctor#1357]
Looking at the code it seems that when block image is used i.e.

```
image::Image1.png[]
```

then obtaining image has to be through `node.attr 'target'` (a Block node), but for inline images

```
image:Image2.png[]
```

it comes directly through a `node.target` (an Inline block).

I reckonf it would be better if this is handled in a consistent manner as it can be very confusing for converter implementors.

**Comment by mojavelinux** [comment]
Yep, I'm well aware of this limitation (as I've often pondered about it). I suppose we could add a target method to AbstractNode and have it implemented to resolve the 'target' attribute in AbstractBlock.

**Comment by mojavelinux** [comment]
Do you think the right approach is to add a target property to the Block class (even though it would only apply for media macros)?

This loosely relates to #2979.

**Comment by ysb33r** [comment]
Yes, I reckon so.
